src/modules/boardApi.js

```diff
@@ -0,0 +1,77 @@
+// Helper function to handle API errors
+const handleApiError = (error) => {
+  // Handle API request error
+  console.error('An error occurred:', error);
+};
+
+// Helper function to make API requests
+const makeApiRequest = async (url, method, body) => {
+  try {
+    const response = await fetch(url, {
+      method,
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify(body),
+    });
+
+    if (!response.ok) {
+      throw new Error('API request failed');
+    }
+
+    const { result } = await response.json();
+    return result;
+  } catch (error) {
+    handleApiError(error);
+    throw error; // Re-throw the error to be handled by the caller
+  }
+};
+
+// Create a new game and save the game ID
+export const createGame = async (gameName) => {
+  try {
+    const response = await makeApiRequest('https://us-central1-js-capstone-backend.cloudfunctions.net/api/games/', 'POST', {
+      name: gameName,
+    });
+
+    const gameId = response.split(' ')[3];
+    localStorage.setItem('gameId', gameId);
+  } catch (error) {
+    handleApiError(error);
+    throw error; // Re-throw the error to be handled by the caller
+  }
+};
+
+// Retrieve all scores for the game
+export const retrieveScores = async () => {
+  const gameId = localStorage.getItem('gameId');
+
+  try {
+    const response = await makeApiRequest(`https://us-central1-js-capstone-backend.cloudfunctions.net/api/games/${gameId}/scores/`, 'GET');
+    return response;
+  } catch (error) {
+    handleApiError(error);
+    return []; // Return an empty array to indicate no scores available
+  }
+};
+
+// Save a score for the game
+export const addYourScore = async (userName, score) => {
+  const gameId = localStorage.getItem('gameId');
+
+  // Validate user input
+  if (!userName || !score || Number.isNaN(parseInt(score, 10))) {
+    // Invalid input, handle the error appropriately
+    return;
+  }
+
+  try {
+    await makeApiRequest(`https://us-central1-js-capstone-backend.cloudfunctions.net/api/games/${gameId}/scores/`, 'POST', {
+      user: userName,
+      score: parseInt(score, 10),
+    });
+  } catch (error) {
+    handleApiError(error);
+    throw error; // Re-throw the error to be handled by the caller
+  }
+};
```

Comment:
- [ ] @Zuwairanajma Great work on your app, however, you are not storing scores added to your API, the makeAPiRequest just fetches the data stored in API  but in this case, you are trying to use it to store the score which make the scores not to be stored when Ithe app is run.